src/frontend/dsl/mod.rs

```diff
@@ -310,6 +310,18 @@ impl<'a, F> StepTypeSetupContext<'a, F> {
     }
 }
 
+impl<'a, F: Eq + PartialEq + Hash + Debug + Clone> StepTypeSetupContext<'a, F> {
+    pub fn auto(&mut self, signal: Queriable<F>, expr: ASTExpr<F>) {
+        self.step_type.auto_signals.insert(signal, expr);
+    }
+
```

Comment:
Will `auto` ever be used alone? If not, can make it `fn`.

Reply:
Yes, auto is like <-- in circom.